framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -266,6 +275,67 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
   }
 }
 
+void
+MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Algorithm Overview:
+  1) Loop through all elements
+  2) For each element loop through all it's sides
+  3) If it has no neighbors it's an external side
+  4) If external check if it's part of a sideset
+  */
+  if (mesh->mesh_dimension() < 2)
+    mooseError("The sideset check only works for 2D and 3D meshes");
+  auto & boundary_info = mesh->get_boundary_info();
+  boundary_info.build_side_list();
+  const auto sideset_map = boundary_info.get_sideset_map();
+  unsigned int num_faces_without_sideset = 0;
+
+  for (const auto elem : mesh->active_element_ptr_range())
+  {
+    std::vector<std::unique_ptr<Elem>> elem_sides(elem->n_sides());
+    for (auto i : elem->side_index_range())
+    {
+      // Check if side is external
+      if (elem->neighbor_ptr(i) == nullptr)
+      {
+        // If external check if that side had a sideset
+        auto side_range = sideset_map.equal_range(elem);
+        bool has_boundary = false;
+        for (const auto & itr : as_range(side_range))
+        {
+          if (itr.second.first == i)
+          {
+            has_boundary = true;
+          }
+        }
+        if (!has_boundary)
+        {
+          // This side does not have a sideset!!!
+          std::string message;
+          if (mesh->mesh_dimension() == 3)
+            message = "Element " + std::to_string(elem->id()) +
+                      " contains an external face which has not been assigned to a sideset";
+          else
+            message = "Element " + std::to_string(elem->id()) +
+                      " contains an external edge which has not been assigned to a sideset";
+          _console << message << std::endl;
```

Comment:
I d be worried that this is too much. A missing boundary might contains thousands of sides

let's add a parameter that is unsigned int and is called "default_number_logs", defaults to 10, that can be used to limit the number of times these things are printed. 
It applied to every diagnostics

Reply:
Good idea. I was looking at the mesh diagnostic documentation and it looks like there's a similar parameter called log_length_limit that defaults to 10. Unless there are cases where we want different log lengths for different diagnostic checks I could probably just reuse that one?

Reply:
nice. i did not know I had added that already

Reply:
maybe someone else did